flow/include/flow/OTELMetrics.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * OTELMetrics.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef FLOW_OTELMETRIC_H
+#define FLOW_OTELMETRIC_H
+#include "flow/flow.h"
+#include <vector>
+
+class Attribute {
+public:
+	std::string key;
+	std::string value;
+
+	Attribute(const std::string& k, const std::string& v) : key{ k }, value{ v } {}
+	Attribute(std::string&& k, std::string&& v) : key{ std::move(k) }, value{ std::move(v) } {}
+};
+
+enum AggregationTemporality {
+	AGGREGATION_TEMPORALITY_UNSPECIFIED = 0,
+	AGGREGATION_TEMPORALITY_DELTA,
+	AGGREGATION_TEMPORALITY_CUMULATIVE
+};
+
+enum DataPointFlags { FLAG_NONE = 0, FLAG_NO_RECORDED_VALUE };
+
+class NumberDataPoint {
+public:
+	double startTime;
+	double recordTime;
+	std::vector<Attribute> attributes;
+	int64_t val;
```

Comment:
The proto definition allows this to be a double as well. Can it ever be a double in FDB?

Reply:
I don't think that's possible because `ICounter::getValue` returns` int64_t`.